my/location/fallback/common.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+from dataclasses import dataclass
+from typing import Optional, Callable, Sequence, Iterator, List, Union
+from datetime import datetime, timedelta, timezone
+
+from ..common import LocationProtocol, Location
+DateExact = Union[datetime, float, int]  # float/int as epoch timestamps
+
+@dataclass
+class FallbackLocation(LocationProtocol):
+    lat: float
+    lon: float
+    dt: datetime
+    duration: Optional[float] = None # time in seconds for how long this is valid
```

Comment:
just a nit, but I usually do something like `Second = float` and then use `Optional[Second]` -- requires less comments too :) and some potential for static analysis later